frame/core-fellowship/src/lib.rs

```diff
@@ -0,0 +1,577 @@
+			new_rank: Option<u16>,
+		},
+		/// Pre-ranked account has been inducted at their current rank.
+		Imported { who: T::AccountId, rank: RankOf<T, I> },
+	}
+
+	#[pallet::error]
+	pub enum Error<T, I = ()> {
+		/// Member's rank is too low.
+		Unranked,
+		/// Member's rank is not zero.
+		Ranked,
+		/// Member's rank is not as expected - generally means that the rank provided to the call
+		/// does not agree with the state of the system.
+		UnexpectedRank,
+		/// The given rank is invalid - this generally means it's not between 1 and 9.
+		InvalidRank,
+		/// The origin does not have enough permission to do this operation.
+		NoPermission,
+		/// No work needs to be done at present for this member.
+		NothingDoing,
+		/// The candidate has already been inducted. This should never happen since it would
+		/// require a candidate (rank 0) to already be tracked in the pallet.
+		AlreadyInducted,
+		/// The candidate has not been inducted, so cannot be offboarded from this pallet.
+		NotTracked,
+		/// Operation cannot be done yet since not enough time has passed.
+		TooSoon,
+	}
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Bump the state of a member.
+		///
+		/// This will demote a member whose `last_proof` is now beyond their rank's
+		/// `demotion_period`.
+		///
+		/// - `origin`: A `Signed` origin of an account.
+		/// - `who`: A member account whose state is to be updated.
+		#[pallet::weight(T::WeightInfo::bump_offboard().max(T::WeightInfo::bump_demote()))]
+		#[pallet::call_index(0)]
+		pub fn bump(origin: OriginFor<T>, who: T::AccountId) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			let mut member = Member::<T, I>::get(&who).ok_or(Error::<T, I>::NotTracked)?;
+			let rank = T::Members::rank_of(&who).ok_or(Error::<T, I>::Unranked)?;
+
+			let params = Params::<T, I>::get();
+			let demotion_period = if rank == 0 {
+				params.offboard_timeout
+			} else {
+				let rank_index = Self::rank_to_index(rank).ok_or(Error::<T, I>::InvalidRank)?;
+				params.demotion_period[rank_index]
+			};
+			let demotion_block = member.last_proof.saturating_add(demotion_period);
+
+			// Ensure enough time has passed.
+			let now = frame_system::Pallet::<T>::block_number();
+			if now >= demotion_block {
+				T::Members::demote(&who)?;
+				let maybe_to_rank = T::Members::rank_of(&who);
+				Self::dispose_evidence(who.clone(), rank, maybe_to_rank);
+				let event = if let Some(to_rank) = maybe_to_rank {
+					member.last_proof = now;
+					Member::<T, I>::insert(&who, &member);
+					Event::<T, I>::Demoted { who, to_rank }
+				} else {
+					Member::<T, I>::remove(&who);
+					Event::<T, I>::Offboarded { who }
+				};
+				Self::deposit_event(event);
+				return Ok(Pays::No.into())
+			}
+
+			Err(Error::<T, I>::NothingDoing.into())
+		}
+
+		/// Set the parameters.
+		///
+		/// - `origin`: A origin complying with `ParamsOrigin` or root.
```

Comment:
Where is the advantage to explicitly include root instead of letting the runtime dev use `EitherOf`?

Reply:
In the case that the call is uninvokable due to origin restrictions we added the [Weightless](https://github.com/paritytech/substrate/pull/12829) error return type which can idiomatically used like:  
```rust
T::CustomOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
```
But keeping it is also fine - it ensures that the benchmarks run.

Reply:
`try_successful_origin` isn't always possible to implement.